spec/function.dd

```diff
@@ -1230,35 +1230,61 @@ int foo(in int x, out int y, ref int z, int q);
     $(THEAD Storage Class, Description)
     $(TROW $(I none), parameter becomes a mutable copy of its argument)
 
-    $(TROW $(D in), defined as $(D scope const).  However $(D in) has not yet been properly
+    $(TROW $(D in), Defined as $(D scope const).  However $(D in) has not yet been properly
     implemented so it's current implementation is equivalent to $(D const).  It is recommended
     to avoid using $(D in) until it is properly defined and implemented.  Use $(D scope const)
     or $(D const) explicitly instead.)
-    $(TROW $(D out), parameter is initialized upon function entry with the default value
+    $(TROW $(D ref), parameter is passed by reference)
+    $(TROW $(D out), parameter is passed by reference and initialized upon function entry with the default value
     for its type)
 
-    $(TROW $(D ref),   parameter is passed by reference)
-    $(TROW $(D scope), references in the parameter
+    $(TROW $(D scope), References to the parameter
     cannot be escaped (e.g. assigned to a global variable).
-    Ignored for parameters with no references)
+    Ignored for parameters with no references.
+    Note: The `scope` attribute is currently only enforced by `dmd`
+    when the `-dip1000` switch is passed.
```

Comment:
Since DIP 25 and DIP 1000 are experimental features (at least I think they are). I'd prefer if we had separate a separate document in the spec describing their semantics.  Then when they are official, their documentation can be folded into the spec.

Reply:
`-dip25` is already in the docs (I wrote it): 
https://dlang.org/spec/function.html#return-ref-parameters

Reply:
Thanks. We're lucky to have it.  But I think it should be moved until DIP 25 and 1000 are official.